src/runtime/containerd-shim-v2/delete.go

```diff
@@ -9,25 +9,21 @@ import (
 	"context"
 	"path"
 
+	"github.com/containerd/containerd/api/types/task"
 	"github.com/containerd/containerd/mount"
 	"github.com/kata-containers/kata-containers/src/runtime/pkg/katautils"
-	"github.com/kata-containers/kata-containers/src/runtime/virtcontainers/types"
 )
 
 func deleteContainer(ctx context.Context, s *service, c *container) error {
-	status, err := s.sandbox.StatusContainer(c.id)
-	if err != nil && !isNotFound(err) {
-		return err
-	}
-	if !c.cType.IsSandbox() && err == nil {
-		if status.State.State != types.StateStopped {
-			_, err = s.sandbox.StopContainer(c.id, false)
+	if !c.cType.IsSandbox() {
```

Comment:
Actually the VC status would be updated here https://github.com/kata-containers/kata-containers/blob/2.0-dev/src/runtime/containerd-shim-v2/wait.go#L71 , Maybe I missed something, I didn't see there is any difference between the VC status and shimv2 status.

Reply:
If `StopContainer` fails, VC status and shimv2 status will be different. It is just an exceptional case.